### ISSUE TITLE
.debug_line dir and file table malformed

### ISSUE THREAD

**Issue by bjorn3** [bjorn3/rustc_codegen_cranelift#308]
```
$ git rev-parse HEAD
058c7ed46a825d09bcfa90db95a1e2f4fb43414e
$ git branch
[...]
* wip_fix_libstd
[...]
$ readelf -w target/out/alloc_example.dummy_name.rcgu.o
[...]
The Directory Table (offset 0x1b):
  1     example
  2     /home/bjorn/Documenten/rustc_codegen_cranelift/build_sysroot/sysroot_src/src/liballoc
  3     /home/bjorn/Documenten/rustc_codegen_cranelift/build_sysroot/sysroot_src/src/libcore
  4     /home/bjorn/Documenten/rustc_codegen_cranelift/build_sysroot/sysroot_src/src/libcore/ops
  5     /home/bjorn/Documenten/rustc_codegen_cranelift/build_sysroot/sysroot_src/src/libstd/thread
  6     /home/bjorn/Documenten/rustc_codegen_cranelift/build_sysroot/sysroot_src/src/libstd/sys/windows
  7     /home/bjorn/Documenten/rustc_codegen_cranelift/build_sysroot/sysroot_src/src/libstd/sys/sgx/abi
  8     /home/bjorn/Documenten/rustc_codegen_cranelift/build_sysroot/sysroot_src/src/libcore/fmt
  9     /home/bjorn/Documenten/rustc_codegen_cranelift/build_sysroot/sysroot_src/src/libcore/num
  10    /home/bjorn/Documenten/rustc_codegen_cranelift/build_sysroot/sysroot_src/src/libstd

 The File Name Table (offset 0x349):
  Entry Dir     Time    Size    Name
  1     47      104     111     /home/bjorn/Documenten/rustc_codegen_cranelift/build_sysroot/sysroot_src/src/libstd/sys/unix/process
  2     47      104     111     me/bjorn/Documenten/rustc_codegen_cranelift/build_sysroot/sysroot_src/src/liballoc/collections/btree
  3     0       97      108     me/bjorn/Documenten/rustc_codegen_cranelift/build_sysroot/sysroot_src/src/liballoc/collections
  4     1       0       0       loc_example.rs
  5     2       0       0       rc.rs
  6     3       0       0       cell.rs
  7     3       0       0       mem.rs
  8     3       0       0       ptr.rs
  9     4       0       0       function.rs
  10    3       0       0       option.rs
  11    5       0       0       local.rs
  12    6       0       0       os_str.rs
  13    7       0       0       mod.rs
  14    7       0       0       reloc.rs
  15    6       0       0       pipe.rs
  16    8       0       0       mod.rs
  17    2       0       0       alloc.rs
  18    9       0       0       f64.rs
  19    3       0       0       alloc.rs
  20    9       0       0       mod.rs
  21    6       0       0       path.rs
  22    10      0       0       rt.rs
  23    11      0       0       <::std::thread::local::__thread_local_inner macros>
  24    2       0       0       boxed.rs
  25    3       0       0       result.rs
  26    3       0       0       macros.rs
  27    10      0       0       process.rs
  28    12      0       0       process_common.rs
  29    3       0       0       panicking.rs
  30    3       0       0       hint.rs
  31    6       0       0       thread.rs
  32    6       0       0       process.rs
[...]
```

**Comment by bjorn3** [comment]
cc @philipc

**Comment by philipc** [comment]
The bug may be that you're adding a zero-length directory (and we don't have an assert in gimli to check for that).

**Comment by bjorn3** [comment]
I should probably match on https://doc.rust-lang.org/nightly/nightly-rustc/syntax_pos/enum.FileName.html

**Comment by bjorn3** [comment]
Fixed in 805af22.

**Comment by bjorn3** [comment]
`<::std::thread::local::__thread_local_inner macros>` was indeed the problem.
